### PR TITLE
Add support for showing the native OS incoming call screen when starting new Element Calls

### PR DESCRIPTION
**Pull request by stefanceriu**
Handle incoming VoIP calls, show the native OS call screen as per https://developer.apple.com/documentation/callkit/sending-end-to-end-encrypted-voip-calls
Defined through https://github.com/matrix-org/matrix-spec-proposals/pull/4075

Fixes https://github.com/element-hq/element-x-ios/issues/2876

Requires https://github.com/matrix-org/matrix-rust-sdk/pull/3434

The way this works is the following:
- the NSE receives the notification and decrypts it
- checks if it's still time relevant (max 10 seconds old) and whether it should ring
- otherwise it goes on to show it as a normal notification
- if it should ring then it discards the notification but invokes `reportNewIncomingVoIPPushPayload`
so that the main app can handle it
- the main app picks this up in `PKPushRegistry.didReceiveIncomingPushWith` and
`CXProvider.reportNewIncomingCall` to show the system UI and handle actions on it.

N.B. this flow works properly only when background processing capabilities are enabled

IMG_2432

### CODE REVIEW

File: ElementX/Sources/Other/HTMLParsing/AttributedStringBuilder.swift [hunk 1]
```diff
@@ -143,6 +143,7 @@ struct AttributedStringBuilder: AttributedStringBuilderProtocol {
         attributedString.removeAttribute(.foregroundColor, range: .init(location: 0, length: attributedString.length))
     }
     
+    // swiftlint:disable:next cyclomatic_complexity
```
Comment (Velin92) [inline ElementX/Sources/Other/HTMLParsing/AttributedStringBuilder.swift:146]: why is this here?

File: ElementX/Sources/Services/ElementCall/ElementCallServiceProtocol.swift [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+}
+
+enum ElementCallServiceNotificationKey: String {
+    case roomID
+    case roomDisplayName
+    case senderID
+    case senderDisplayName
+}
+
+let ElementCallServiceNotificationDiscardDelta = 10.0
```
Comment (Velin92) [inline ElementX/Sources/Services/ElementCall/ElementCallServiceProtocol.swift:31]: maybe let's put this into a `ElementCallServiceConstants` enum as a `static let`

**Review by Velin92 — APPROVED** [review]
LGTM just some suggestions

File: ElementX/Sources/Other/HTMLParsing/AttributedStringBuilder.swift (hunk 1 above)
Comment (stefanceriu) [reply ElementX/Sources/Other/HTMLParsing/AttributedStringBuilder.swift:146]: Because we have a warning there? 🤔 

Screenshot 2024-05-22 at 10 40 07

File: ElementX/Sources/Services/ElementCall/ElementCallServiceProtocol.swift (hunk 2 above)
Comment (stefanceriu) [reply ElementX/Sources/Services/ElementCall/ElementCallServiceProtocol.swift:31]: Why would that be better? 🤔

**Review by Velin92 — APPROVED** [review]
